app/presenters/confirmation_letter_presenter.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+class ConfirmationLetterPresenter
+  delegate :reference, to: :registration
+
+  def initialize(registration)
+    @registration = registration
+  end
+
+  def sorted_active_registration_exemptions
+    registration_exemptions_with_exemptions.where(state: :active).order(:exemption_id)
+  end
+
+  def sorted_deregistered_registration_exemptions
+    registration_exemptions_with_exemptions.where("state != ?", :active).order_by_state_then_exemption_id
+  end
+
+  # Used only if the page contains a single letter.
+  def webpage_title
+    I18n.t("confirmation_letter.single_title", reg_number: registration.reference)
+  end
+
+  def date_of_letter
+    I18n.l(Time.zone.today, format: :long)
+  end
+
+  # Provides the full postal address for the letter.
+  def postal_address_lines
+    [
+      "#{registration.contact_first_name} #{registration.contact_last_name}",
+      registration.operator_name,
+      address_lines(registration.contact_address)
+    ].flatten!.reject(&:blank?)
+  end
+
+  def reg_details_items
+    submission_date = I18n.l(registration.submitted_at.to_date, format: :long)
+    filter_blank_items([
+                         { key: t("reg_details_reference"), value: registration.reference },
+                         { key: t("reg_details_activation_date"), value: submission_date }
+                       ])
+  end
+
+  def reg_completed_by_items
+    applicant_full_name = "#{registration.applicant_first_name} #{registration.applicant_last_name}"
+    filter_blank_items([
+                         { key: t("reg_completed_by_name"), value: applicant_full_name },
+                         { key: t("reg_completed_by_telephone"), value: registration.applicant_phone },
+                         { key: t("reg_completed_by_email"), value: registration.applicant_email }
+                       ])
+  end
+
+  def organisation_items
+    filter_blank_items([
+      { key: t("organisation_type"), value: I18n.t(registration.business_type, scope: "organisation_type") },
```

Comment:
This seems to cause a translation error for some business types – I think we only have text in the YAML file for limited companies and LLPs. Not sure why!

Reply:
This is fixed in 235b1ab.